lib/src/rpc/dial.dart

```diff
@@ -129,16 +134,23 @@ Future<ClientChannelBase> dial(String address, DialOptions? options, String Func
     } catch (e) {
       _logger.d('Error dialing with mDNS; falling back to other methods', error: e);
     }
+    mdnsSW.stop();
+    _logger.d('STATS: mDNS discovery took ${mdnsSW.elapsed}');
   }
 
   bool disableWebRtc = opts.webRtcOptions?.disable ?? false;
   if (address.contains('.local.') || address.contains('localhost')) {
     disableWebRtc = true;
   }
+  final chan;
   if (disableWebRtc) {
-    return _dialDirectGrpc(address, opts, sessionCallback);
+    chan = _dialDirectGrpc(address, opts, sessionCallback).timeout(opts.timeout);
```

Comment:
I sense that this might not be the idiomatic way to provide a timeout to a `Future`, @clintpurser / @njooma . I went down quite a rabbit-hole of using `Completer`s to try to guard wherever we call `dial` in the SDK, but what I've done here does seem to limit the amount of time we're able to `await` the `Future<ClientChannelBase>`... Hitting the timeout creates a "context deadline exceeded" error that seems to break app, though, so would love to talk offline about how to `catch` that and gracefully report a timeout in connection.

Reply:
Cool; I think an `onTimeout` function will likely be the way to go. Looking into it now, but may need some advice from other Flutterers to get a sense of how to not break the mobile/example apps. Thanks for the input!